core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -550,6 +550,229 @@ public void testDeleteFilesTableSelection() throws IOException {
     Assert.assertEquals(expected, scan.schema().asStruct());
   }
 
+  @Test
+  public void testPartitionSpecEvolutionAdditiveV1() {
```

Comment:
Do we need the separate tests for the version here?

Seems like the only difference is the Asserts?

Reply:
Difference was the way to make PartitionKey.  Combined the test using different formatVersion handling for this part and the asserts.